eng/jobs/windows-build.yml

```diff
@@ -5,188 +5,139 @@ parameters:
-        name: NetCore1ESPool-Internal
-        demands: ImageOverride -equals windows.vs2022.amd64
-    strategy:
-      matrix:
-        release:
-          _BuildConfig: Release
-        ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
-          debug:
-            _BuildConfig: Debug
-    workspace:
-      clean: all
-    variables: 
-      CommonMSBuildArgs: >-
-        -c $(_BuildConfig)
-        /p:TargetArchitecture=${{ parameters.targetArchitecture }}
-        /p:DotNetSignType=$(SignType)
-      ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-        InternalMSBuildArgs: >-
-          /p:OfficialBuildId=$(Build.BuildNumber)
-          /p:DotNetPublishUsingPipelines=true
-          /p:Test=false
-      ${{ else }}:
-        InternalMSBuildArgs: ''
-      Codeql.Enabled: ${{ parameters.codeql }}
-
-    steps:
-
-    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-      - task: NuGetAuthenticate@1
-
-      - task: MicroBuildSigningPlugin@2
-        displayName: Install MicroBuild plugin for Signing
-        inputs:
-          signType: $(SignType)
-          zipSources: false
-          feedSource: https://dnceng.pkgs.visualstudio.com/_packaging/MicroBuildToolset/nuget/v3/index.json
-        continueOnError: false
-        condition: and(succeeded(), in(variables['SignType'], 'real', 'test'))
-
-    # NuGet's http cache lasts 30 minutes. If we're on a static machine, this may interfere with
-    # auto-update PRs by preventing the CI build from fetching the new version. Delete the cache.
-    - powershell: Remove-Item -Recurse -ErrorAction Ignore "$env:LocalAppData\NuGet\v3-cache"
-      displayName: Clear NuGet http cache (if exists)
-
-    - ${{ if ne(variables['System.TeamProject'], 'public') }}:
-      - task: PowerShell@2
-        displayName: Setup Private Feeds Credentials
-        condition: eq(variables['Agent.OS'], 'Windows_NT')
-        inputs:
-          filePath: $(Build.SourcesDirectory)/eng/common/SetupNugetSources.ps1
-          arguments: -ConfigFile $(Build.SourcesDirectory)/NuGet.config -Password $Env:Token
-        env:
-          Token: $(dn-bot-dnceng-artifact-feeds-rw)
-
-    - script: >-
-        eng\common\cibuild.cmd
-        $(CommonMSBuildArgs)
-        $(InternalMSBuildArgs)
-      displayName: Build
-
-    # Generate SBOM for the internal leg only
-    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-      - template: ..\common\templates\steps\generate-sbom.yml
-        parameters:
-          name: Generate_SBOM_${{ parameters.name }}
-
-    # Push VS packages to VS package feed
+- job: ${{ parameters.name }}
+  displayName: ${{ parameters.displayName }}
+  timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+  strategy:
+    matrix:
```

Comment:
Double check this with the 1ES docs. There was talk about stragegy/matrix not always working properly, but I've also seen others that mentioned it worked just fine.

Reply:
It seems to be working fine in my validation builds I linked above.